zkvm/src/vm.rs

```diff
@@ -51,6 +56,99 @@ pub struct Tx {
     pub proof: R1CSProof,
 }
 
+impl Tx {
+    /// Serializes the tx into a byte array.
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut buf = Vec::with_capacity(self.serialized_size());
+        encoding::write_u64(self.header.version, &mut buf);
```

Comment:
Same as with parsing `TxHeader`: we can factor this out into `TxHeader::encode()`.